apis/telemetry/v1beta1/logpipeline_types.go

```diff
@@ -0,0 +1,235 @@
+type Output struct {
+	// Defines a custom output in the Fluent Bit syntax. Note: If you use a `custom` output, you put the LogPipeline in unsupported mode.
+	Custom string `json:"custom,omitempty"`
+	// Configures an HTTP-based output compatible with the Fluent Bit HTTP output plugin.
+	HTTP *HTTPOutput `json:"http,omitempty"`
+	// The grafana-loki output is not supported anymore. For integration with a custom Loki installation, use the `custom` output and follow [Installing a custom Loki stack in Kyma](https://kyma-project.io/#/telemetry-manager/user/integration/loki/README ).
+	Loki *LokiOutput `json:"grafana-loki,omitempty"`
+}
+
+func (i *Input) IsDefined() bool {
+	return i != nil
+}
+
+func (o *Output) IsCustomDefined() bool {
+	return o.Custom != ""
+}
+
+func (o *Output) IsHTTPDefined() bool {
+	return o.HTTP != nil && o.HTTP.Host.IsDefined()
+}
+
+func (o *Output) IsLokiDefined() bool {
+	return o.Loki != nil && o.Loki.URL.IsDefined()
+}
+
+func (o *Output) IsAnyDefined() bool {
+	return o.pluginCount() > 0
+}
+
+func (o *Output) IsSingleDefined() bool {
+	return o.pluginCount() == 1
+}
+
+func (o *Output) pluginCount() int {
+	plugins := 0
+	if o.IsCustomDefined() {
+		plugins++
+	}
+	if o.IsHTTPDefined() {
+		plugins++
+	}
+	if o.IsLokiDefined() {
+		plugins++
+	}
+	return plugins
+}
+
+// Provides file content to be consumed by a LogPipeline configuration
+type FileMount struct {
+	Name    string `json:"name,omitempty"`
+	Content string `json:"content,omitempty"`
+}
+
+// References a Kubernetes secret that should be provided as environment variable to Fluent Bit
+type VariableRef struct {
+	// Name of the variable to map.
+	Name      string          `json:"name,omitempty"`
+	ValueFrom ValueFromSource `json:"valueFrom,omitempty"`
+}
+
+// LogPipelineStatus shows the observed state of the LogPipeline
+type LogPipelineStatus struct {
+	// An array of conditions describing the status of the pipeline.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// Is active when the LogPipeline uses a `custom` output or filter; see [unsupported mode](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/02-logs.md#unsupported-mode).
+	UnsupportedMode bool `json:"unsupportedMode,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Configuration Generated",type=string,JSONPath=`.status.conditions[?(@.type=="ConfigurationGenerated")].status`
+// +kubebuilder:printcolumn:name="Agent Healthy",type=string,JSONPath=`.status.conditions[?(@.type=="AgentHealthy")].status`
+// +kubebuilder:printcolumn:name="Unsupported-Mode",type=boolean,JSONPath=`.status.unsupportedMode`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:storageversion
+
+// LogPipeline is the Schema for the logpipelines API
+type LogPipeline struct {
```

Comment:
Let's make the type declaration order more consistent. I saw the following order in Kubernetes and other opensource projects (for example in https://github.com/kubernetes/api/blob/master/apps/v1/types.go):
- Main type (LogPipeline, MetricPipeline, TracePipeline)
- Spec
- Spec subtypes
- Status
- Status subtypes
- Main type list
- init func